### PR TITLE
Add permits to rda for api

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

File: pkg/apis/dnsalias/rbac.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	clusterRole.Rules[0].ResourceNames = getRoleResourceNamesWithout(dnsAliasName, clusterRole)
+	if len(clusterRole.Rules[0].ResourceNames) == 0 {
+		return s.deleteClusterRoleAndBinding(roleName)
+	}
+
+	clusterRole.ObjectMeta.OwnerReferences = getOwnerReferencesWithout(dnsAliasName, clusterRole)
+	return s.kubeUtil.ApplyClusterRole(clusterRole)
+}
+
+func getOwnerReferencesWithout(dnsAliasName string, clusterRole *rbacv1.ClusterRole) []metav1.OwnerReference {
```
[the remark this answers is not in the document]
Comment (Richard87) [reply pkg/apis/dnsalias/rbac.go:155]: Or something like this:

```go
clusterRole.ObjectMeta.OwnerReferences =  slices.DeleteFunc(clusterRole.ObjectMeta.OwnerReferences, func(o metav1.OwnerReference) bool {
  return o.Name == dnsAliasName
})
```
(https://pkg.go.dev/slices@master#DeleteFunc)

File: pkg/apis/utils/application_builder.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func (ap *ApplicationBuilderStruct) WithBuildCache(useBuildCache *bool) Applicat
 // WithPrivateImageRegistry adds a private image hub to application
 func (ap *ApplicationBuilderStruct) WithPrivateImageRegistry(server, username, email string) ApplicationBuilder {
 	if ap.privateImageHubs == nil {
-		ap.privateImageHubs = map[string]*radixv1.RadixPrivateImageHubCredential{}
+		ap.privateImageHubs = radixv1.PrivateImageHubEntries(map[string]*radixv1.RadixPrivateImageHubCredential{})
```
Comment (Richard87) [inline pkg/apis/utils/application_builder.go:58]: I get a "Redundant type conversion" here

image

**Review by Richard87 — CHANGES_REQUESTED** [review]
just a few nitpicks

File: pkg/apis/dnsalias/rbac.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (satr) [reply pkg/apis/dnsalias/rbac.go]: Fixed

**Review by Richard87 — APPROVED** [review]
lgtm!

**Pull request merged**